connector/mqtt/vernemq/k2v-bridge/app/MqttClient.js

```diff
@@ -69,46 +70,60 @@ class MqttClient {
     const mqttOnConnectBind = this.onConnect.bind(this);
     const mqttOnDisconnectBind = this.onDisconnect.bind(this);
 
-    logger.info('Binding event callbacks', TAG);
+    this.logger.info('Binding event callbacks');
     this.mqttc.on('connect', mqttOnConnectBind);
     this.mqttc.on('disconnect', mqttOnDisconnectBind);
   }
 
   /**
-   * Reached when the MQTTClient
-   * connects successfully to the broker
+   * Reached when the MQTTClient connects successfully to the broker
+   * @access private
    * @callback MQTTClient~onConnect
    */
   onConnect() {
-    logger.info('MQTT connection established', TAG);
+    this.logger.info('MQTT connection established');
     this.isConnected = true;
     this.agentMessenger.init();
   }
 
   /**
    * Reached when the MQTTClient disconnects from the broker.
+   * @access private
    * @callback MQTTClient~onDisconnect
    */
   onDisconnect() {
     this.isConnected = false;
-    logger.info('MQTT connection ended, trying to reconnect...', TAG);
+    this.logger.info('MQTT connection ended, trying to reconnect...');
     this.mqttc.reconnect();
     // TODO: agentMessenger (.pause or .stop)
   }
 
   /**
    * Publishes a message to a given topic.
-   * @param {string} topic
-   * @param {string} message
+   * @access public
+   * @function publishMessage
+   *
+   * @param {*} data message object retrieved from Kafka
    */
-  publishMessage(topic, message) {
-    if (this.isConnected && this.mqttc) {
-      logger.debug(`Publishing on topic ${topic}`, TAG);
-      this.mqttc.publish(topic, message, { qos: this.publishQos });
-    } else {
-      logger.error('Client not connected', TAG);
+  publishMessage(data) {
+    try {
+      if (this.isConnected && this.mqttc) {
+        const value = JSON.parse(data.value.toString());
+
+        const topic = utils.generateDojotActuationTopic(
+          value.meta.service,
+          value.data.id,
+          mqttConfig['client.publish.topic.suffix'],
+        );
+
+        this.mqttc.publish(topic, JSON.stringify(value.data.attrs), { qos: this.publishQos });
```

Comment:
If I'm no wrong this method allows to pass an error callback. It would be nice to log if something unexpected happens.